jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/DispatchedMessageSink.java

```diff
@@ -51,9 +51,9 @@ public abstract class DispatchedMessageSink extends AbstractMessageSink
     private volatile CompletableFuture<Void> dispatchComplete;
     private MessageSink typeSink;
 
-    public DispatchedMessageSink(CoreSession session, MethodHandle methodHandle, boolean autoDemand)
+    public DispatchedMessageSink(CoreSession session, MethodHolder methodHolder, boolean autoDemand)
```

Comment:
Do we need to provide all the old APIs deprecated?
```suggestion
    @Deprecated
    public DispatchedMessageSink(CoreSession session, MethodHandle methodHandle, boolean autoDemand)
    {
        this(session, MethodHolder.from(methodHandle), autoDemand);
    }
    public DispatchedMessageSink(CoreSession session, MethodHolder methodHolder, boolean autoDemand)
```

Reply:
I don't think so, this is websocket-core, and this is really not supposed to be used by the application more of an implementation class.

Reply:
yeah but it is public and not internal.... @sbordet thoughts?

Reply:
I think this probably should be internal.

Its to reduce having code duplication from having this in our jakarta and jetty websocket implementations. But making it internal makes things difficult with JPMS.

But there is no API to actually use a `MessageSink` directly.

Reply:
OK, leave out and we can add back deprecated if anybody complains.... although it is against our new deprecated policy :)